### DISCUSSION TITLE
ModuleNotFoundError

### DISCUSSION THREAD

**Discussion by joyboy-dot** [aidudezzz/deepbots-tutorials#45] [General]
![屏幕截图 2024-11-13 163904](https://github.com/user-attachments/assets/bc68d0d9-9f49-4fa9-a8a5-733ef674cd39)
The error message "No module named 'deepbots.supervisor.controllers.robot_supervisor_env'" means that Python is unable to find the robot_supervisor_env module within the deepbots.supervisor.controllers package.

**Comment by tsampazk** [comment]
Hi @joyboy-dot! It seems you have an old version of deepbots. Deepbots should be at version 1.0.0 when installed through `pip install deepbots`. 

Could you try that?

**Reply by joyboy-dot** [reply]
Thanks for your prompt reply, it is indeed a deepbots version issue. When pip installed deepbots, it told me that the 0.22 version of gym could not match the 1.0.0 version of deepbots, so it replaced me with the old version of deepbots. What version of gym should I install to match the 1.0.0 version of deepbots?

**Reply by tsampazk** [reply]
You are welcome!

It seems weird that it defaulted to an older version of deepbots due to having an incompatible gym, but i think that was a pip mishap.

Unfortunately you need to downgrade gym to 0.21.0. I would suggest to use a virtual environment so you can install whatever is needed cleanly for specific usage, in case you need other gym versions for other projects.

The requirements are set and drawn from the [requirements file here.](https://github.com/aidudezzz/deepbots/blob/dev/requirements.txt )

**Reply by joyboy-dot** [reply]
The installation of dependent packages is solved, but a small bug occurs when running the program. [self.position_sensor = self.getDevice("polePosSensor")] This code does not seem to get polePosSensor, but the world file sets a sensor with this name.
![1](https://github.com/user-attachments/assets/f24da3ac-9a10-441a-a168-559cdabb4728)
![2](https://github.com/user-attachments/assets/0c1eb5e9-780c-41a6-9af5-c066c7ba137c)
![3](https://github.com/user-attachments/assets/3f5df1ee-4e3d-4347-a036-a125653daee7)

**Reply by tsampazk** [reply]
Hey @joyboy-dot, i have just tested the code from the [full project](https://github.com/aidudezzz/deepbots-tutorials/tree/master/robotSupervisorSchemeTutorial/full_project) directory and it seems to work fine (added a print):

![image](https://github.com/user-attachments/assets/a05c3f35-b403-4098-b4a8-8e0e18ba9e1c)


![image](https://github.com/user-attachments/assets/88f13856-8622-4920-98c1-d83b1f4cbc6d)


If you are using the same code, may i ask what version of Webots you are running? Additionally, i fixed the bug with the endpoint scale that made the ball appear quite huge :)

**Comment by joyboy-dot** [comment]
@tsampazk Looking forward to your answers
